### PR TITLE
Adjustment of configuration

### PR DESCRIPTION
**Pull request by yfouquet**
We currently use the default typesafe configuration with fallback on reference.conf and default.conf files.
This creates some issue:
- if a library with a reference.conf file uses consul-client, and if a key is defined in both file, then we cannot know which value will be kept (as reference.conf files are merged).
- if a library with more specific configuration loader uses consul-client, it cannot share its configuration with consul-client (and so it cannot overrides any value).
- finally, static instance of configuration makes the library harder to test.

The issue is fix by:
- injecting a Config in the cache builder (default typesafe loader is used if not set). Note that instead a injecting the CacheConfig, we inject a more generic config to ensure that future configuration change have smaller footprint.
- renamming reference.conf and default.conf files to ensure that no merge will be made.
- remove shading of typesage (as exposed in the public API).

A nice side effect is that we can know set different configuration in several Consul instances.

### CODE REVIEW

File: src/main/java/com/orbitz/consul/BaseClient.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.orbitz.consul;
+
+import com.orbitz.consul.config.ConsulConfig;
+
+abstract class BaseClient {
+
+    private ConsulConfig config;
```
Comment (adericbourg) [inline src/main/java/com/orbitz/consul/BaseClient.java]: Should be `final`

**Comment by rickfast** [comment]
I'm out of the country at the moment and won't have time to look at this until next week. My gut feeling (without actually looking at what you've done) is that I don't really want the client to depend on a specific configuration mechanism. I believe this should be the responsible of the client code using the library.

**Comment by yfouquet** [comment]
I do agree that configuration should depend on client code; that's exactly the aim of this PR. By injecting the Config object, their is no "magic" mechanism to understand.
I'll let you check that when you will have time.

**Comment by yfouquet** [comment]
After discussion with my team, i'll make a new PR without typesafe.

**Pull request closed without merging**